internal/pkg/cli/svc_init.go

```diff
@@ -432,6 +431,20 @@ func parseHealthCheck(df dockerfileParser) (*manifest.ContainerHealthCheck, erro
 	}, nil
 }
 
+func redirectPlatform(engine dockerEngine, image string) (string, error) {
+	_, arch, err := dockerPlatform(engine, image)
+	if err != nil {
+		return "", err
+	}
+	// Log a message informing non-default arch users of platform for build.
+	if arch != exec.Amd64Arch {
+		log.Warningf("Architecture type %s is currently unsupported. Setting platform %s instead.\n", arch, fmt.Sprintf(fmtOSArch, exec.LinuxOS, exec.Amd64Arch))
+		return fmt.Sprintf(fmtOSArch, exec.LinuxOS, exec.Amd64Arch), nil
+	}
+	// For now, the above enables ARM archs; other non-default platforms will proceed as before.
+	return "", nil
+}
+
 func dockerPlatform(engine dockerEngine, image string) (os, arch string, err error) {
 	os, arch = runtime.GOOS, runtime.GOARCH
```

Comment:
I think it shouldn't be the responsibility of this function to check if `image` is empty, can we instead merge the two functions into only `redirectPlatform`:
```go
if image == "" {
   return "", nil // Nothing to do.
}
os, arch, err := engine.GetPlatform()
if err != nil { ... }

if arch != exec.Amd64Arch { ... }
return "", nil
```

Reply:
Wait, you mean `if image != ""`, right? 🤔 
That's the case in which we wouldn't build so the `--platform` flag is irrelevant.

Reply:
Yea sorry 😅